membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -197,12 +198,15 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext, syste
 
 object AttributesFromZuora {
   def mergeDigitalSubscriptionExpiryDate(regular: Option[ZuoraAttributes], gift: Option[ZuoraAttributes]) = {
-    val mergedDigisubExpiry = List(regular.map(_.DigitalSubscriptionExpiryDate), gift.map(_.DigitalSubscriptionExpiryDate))
-      .flatten.flatten
-      .sortWith((first, second) => first.isAfter(second))
-      .headOption
+    val maybeExpiryDates = Option(
+      List(regular, gift).flatten.flatMap(_.DigitalSubscriptionExpiryDate)
+    ).filter(_.nonEmpty)
 
-    regular.map(_.copy(DigitalSubscriptionExpiryDate = mergedDigisubExpiry)).orElse(gift)
+    val maybeLatestDate = maybeExpiryDates.map(
+      _.maxBy(_.toDateTimeAtStartOfDay.getMillis)
+    )
```

Comment:
we're getting in to diminishing returns by now, but I have no idea why maxBy doesn't return an Option, as we seem to have more lines of code than using sortWith (although clearer)
do you think it would be clearer to implement a version of maxBy that returns an option, and using that in both places?

Reply:
Tbh I don't think it's worth it, we've been through about 3 different formulations of this code now let's stick with what we've got.